### PR TITLE
refactor: make underlying default

### PR DESCRIPTION
**Pull request by sakulstra**

### CODE REVIEW

File: src/StaticATokenLM.sol [hunk 1]
```diff
@@ -491,24 +475,23 @@ contract StaticATokenLM is
   function _withdraw(
     address owner,
     address receiver,
-    uint256 staticAmount,
```
Comment (sakulstra) [inline src/StaticATokenLM.sol:494]: renamign to be more inline with the rest of the wording

File: src/StaticATokenLM.sol [hunk 2]
```diff
@@ -457,20 +439,33 @@ contract StaticATokenLM is
     address owner,
     bool toUnderlying
   ) external virtual returns (uint256, uint256) {
-    require(shares <= maxRedeem(owner), 'ERC4626: redeem more than max');
-
     return _withdraw(owner, receiver, shares, 0, toUnderlying);
   }
 
   function _deposit(
```
Comment (sakulstra) [inline src/StaticATokenLM.sol:445]: I'm not so sure about this change - essentially aligned with `_withdraw`

**Comment by eboadom** [comment]
@sakulstra I think we should also rename the `fromUnderlying` and `toUnderlying` to something clearer, and respecting how the interface worked.
Maybe `depositToAave` and `withdrawFromAave`?
The rationale is to keep pointing out integrators to the non-4626 functions, and having that interface not losing compatibility

File: src/StaticATokenLM.sol [hunk 3]
```diff
@@ -340,28 +337,20 @@ contract StaticATokenLM is
...
-  ///@inheritdoc IERC4626
-  function maxDeposit(address) public view virtual returns (uint256) {
-    return type(uint256).max;
-  }
-
   ///@inheritdoc IERC4626
   function maxMint(address) public view virtual returns (uint256) {
-    return type(uint256).max;
+    uint256 assets = maxDeposit(address(0));
+    return assets == 0 ? 0 : _convertToShares(assets, Rounding.DOWN);
```
Comment (kyzia551) [inline src/StaticATokenLM.sol:343]: I would either remove this ternary operator completely or move it inside _convertToShares. But better to remove imo

File: src/StaticATokenLM.sol [hunk 4]
```diff
@@ -340,28 +337,20 @@ contract StaticATokenLM is
...
   }
 
   ///@inheritdoc IERC4626
   function maxWithdraw(address owner) public view virtual returns (uint256) {
-    return _convertToAssets(balanceOf[owner], Rounding.DOWN);
+    uint256 shares = maxRedeem(owner);
+    return shares == 0 ? 0 : _convertToAssets(shares, Rounding.DOWN);
```
Comment (kyzia551) [inline src/StaticATokenLM.sol:349]: I would either remove this ternary operator completely or move it inside _convertToAssets. But better to remove IMO. Now does not look consistent with other usages

**Review by kyzia551 — CHANGES_REQUESTED** [review]

**Review by kyzia551 — APPROVED** [review]

**Pull request merged**